lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -3223,15 +3259,44 @@ private long prepareCommitInternal() throws IOException {
               // sneak into the commit point:
               toCommit = segmentInfos.clone();
 
+              if (anyChanges) {
+                mergeAwaitLatchRef = new AtomicReference<>();
+                MergePolicy mergeOnCommitPolicy = waitForMergeOnCommitPolicy(config.getMergePolicy(), toCommit, mergeAwaitLatchRef);
+
+                // Find any merges that can execute on commit (per MergePolicy).
+                commitMerges = mergeOnCommitPolicy.findCommitMerges(segmentInfos, this);
+                if (commitMerges != null && commitMerges.merges.size() > 0) {
```

Comment:
Would we sometimes get !=null and == 0??

Reply:
Based on e.g. NoMergePolicy, it looks like the convention to say "I found no merges" is to return null, but I can't see anything preventing a MergePolicy from returning an empty MergeSpecification.